rastervision_core/rastervision/core/box.py

```diff
@@ -21,27 +22,28 @@ class BoxSizeError(ValueError):
     pass
 
 
-class Box():
+class Box(BaseModel):
     """A multi-purpose box (ie. rectangle) representation ."""
 
-    def __init__(self, ymin, xmin, ymax, xmax):
-        """Construct a bounding box.
-
-        Unless otherwise stated, the convention is that these coordinates are
-        in pixel coordinates and represent boxes that lie within a
-        RasterSource.
-
-        Args:
-            ymin: minimum y value (y is row)
-            xmin: minimum x value (x is column)
-            ymax: maximum y value
-            xmax: maximum x value
-
-        """
-        self.ymin = ymin
-        self.xmin = xmin
-        self.ymax = ymax
-        self.xmax = xmax
+    # yapf: disable
+    ymin: Union[float, int] = Field(..., description='minimum y value (y is row)')
+    xmin: Union[float, int] = Field(..., description='minimum x value (x is column)')
+    ymax: Union[float, int] = Field(..., description='maximum y value')
+    xmax: Union[float, int] = Field(..., description='maximum x value')
+    # yapf: enable
+
+    def __init__(self,
+                 ymin: Optional[Union[int, float]] = None,
+                 xmin: Optional[Union[int, float]] = None,
+                 ymax: Optional[Union[int, float]] = None,
+                 xmax: Optional[Union[int, float]] = None,
+                 **data):
+
+        super().__init__(ymin=ymin, xmin=xmin, ymax=ymax, xmax=xmax, **data)
+        for field in ['ymin', 'xmin', 'ymax', 'xmax']:
+            value = getattr(self, field)
+            if isinstance(value, float) and value.is_integer():
+                setattr(self, field, int(value))
 
```

Comment:
Parts of the code strongly assume `Box` has integer fields (e.g. fields are used for array indices) whereas parts assume that `Box` has float fields (e.g. CRS calculations using extents).  This ensures that particular fields have type `int` if they are whole numbers.